torch/distributions/multivariate_normal.py

```diff
@@ -62,7 +72,7 @@ def _batch_mahalanobis(L, x):
     Accepts batches for both L and x.
     """
     # TODO: use `torch.potrs` or similar once a backwards pass is implemented.
-    flat_L = L.unsqueeze(0).contiguous().view((-1,) + L.shape[-2:])
+    flat_L = L.unsqueeze(0).reshape((-1,) + L.shape[-2:])
```

Comment:
Are you sure this has the correct behavior for non-contiguous tensors?

Reply:
Yes. Reshape first checks if the tensor is "viewable" to the view size. If not, it clones the tensor before doing a `view`. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/TensorShape.cpp#L215-L224